pkgs/development/python-modules/hatchling/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+
+# All runtime dependencies
+, editables
+, importlib-metadata
+, packaging
+, pathspec
+, pluggy
+, tomli
+}:
+
+buildPythonPackage rec {
+  pname = "hatchling";
+  version = "0.19.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "942b34e454ce3be5b03139e0a198ba8ebc08a6095d345fe829b1ae92e9326c2c";
+  };
+
+  propagatedBuildInputs = [
+    editables
+    packaging
+    pathspec
+    pluggy
+    tomli
+  ] ++ lib.optionals (pythonOlder "3.8") [
+    importlib-metadata
+  ];
+
+  pythonImportsCheck = [ "hatchling.build" ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Modern, extensible Python build backend";
+    homepage = "https://github.com/ofek/hatch";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ofek ];
```

Comment:
a entry for `ofek` has to be added to `./maintainers/maintainer-list.nix`
https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers
the `maintainers: add ofek` commit should be before the init commit